spec/features/guests/show_spec.rb

```diff
@@ -0,0 +1,151 @@
+      within("div#room-#{@room2.id}") do
+        expect(page).to have_content(@room2.suite)
+        expect(page).to have_content(@room2.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+
+      expect(page).to have_css("div#room-#{@room3.id}")
+      within("div#room-#{@room3.id}") do
+        expect(page).to have_content(@room3.suite)
+        expect(page).to have_content(@room3.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+    end
+    
+    #Guest 2 Room History
+    visit "/guests/#{@guest2.id}"
+    within("div#guest_room_history") do
+      expect(page).to have_css("div#room-#{@room1.id}")
+      within("div#room-#{@room1.id}") do
+        expect(page).to have_content(@room1.suite)
+        expect(page).to have_content(@room1.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+      
+      expect(page).to_not have_css("div#room-#{@room2.id}")
+      
+      expect(page).to_not have_css("div#room-#{@room3.id}")
+    end
+
+    # Guest 3 Room History
+    visit "/guests/#{@guest3.id}"
+    within("div#guest_room_history") do
+      expect(page).to_not have_css("div#room-#{@room1.id}")
+    
+      expect(page).to have_css("div#room-#{@room2.id}")
+      within("div#room-#{@room2.id}") do
+        expect(page).to have_content(@room2.suite)
+        expect(page).to have_content(@room2.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+    
+      expect(page).to_not have_css("div#room-#{@room3.id}")
+    end
+  end
+  # =================
+  # END STORY 1 TESTS
+  # =================
+
+  # ====================================
+  # STORY 2: ADD A GUEST TO A ROOM TESTS
+  # ====================================
+
+  it "I see a form to add a room to this guest" do
+    visit "/guests/#{@guest2.id}"
+
+    within("div#add-room") do
+      expect(page).to have_css("form")
+      expect(page).to have_field(:room_id, type: "text")
+    end
+  end
+
+  describe "When I fill in a field with the ID of an existing room and click 'submit'" do
+    it "Then I am redirected back to the guest's show page see the room now listed under this guest's rooms" do
+      visit "/guests/#{@guest2.id}"
+      
+      expect(@guest2.rooms).to_not include(@room2)
+      within("div#guest_room_history") do
+        expect(page).to_not have_css("div#room-#{@room2.id}")
+      end
+
+      within("div#add-room") do
+        fill_in "room_id", with: "#{@room2.id}"
+        click_button "Submit"
+      end
+
+      expect(current_path).to eq("/guests/#{@guest2.id}")
+
+      within("div#guest_room_history") do
+        expect(page).to have_css("div#room-#{@room2.id}")
+        within("div#room-#{@room2.id}") do
+          expect(page).to have_content(@room2.suite)
+          expect(page).to have_content(@room2.rate)
+          expect(page).to have_content(@hotel.name)
+        end
+      end
+
+      @guest2.reload
+      expect(@guest2.rooms).to include(@room2)
```

Comment:
This test works great, except doing `.to include(...)` is not really testing that it's on hte page - remember a feature test is looking at the rendered view specifically, not at objects. So, I would rewrite this and line 121 to be
```
expect(page).to have_content(@room2.suite)
```

Reply:
I have `expect(page).to have_content(@room2.suite)` on line 136 and sort of on line 123 
_(123 tests for the css <div> block, but I probably should have tested for content)_